doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/docUtils.kt

```diff
@@ -0,0 +1,404 @@
+        var line = it
+
+        if (i == 0) {
+            line = line.trimStart().removePrefix("/**")
+        }
+        if (i == lines.count() - 1) {
+            val lastLine = line.trimStart()
+
+            line = if (lastLine == "*/") {
+                ""
+            } else {
+                lastLine
+                    .removePrefix("*")
+                    .removeSuffix("*/")
+                    .removeSuffix(" ") // optional extra space at the end
+            }
+        }
+        if (i != 0 && i != lines.count() - 1) {
+            line = line.trimStart().removePrefix("*")
+        }
+
+        line = line.removePrefix(" ") // optional extra space at the start
+
+        line
+    }
+
+    return result
+}
+
+
+/**
+ * Turns multi-line String into valid KDoc/Javadoc.
+ */
+fun DocContent.toDoc(indent: Int = 0): String =
+    this
+        .split('\n')
+        .toMutableList()
+        .let {
+            it[0] = if (it[0].isEmpty()) "/**" else "/** ${it[0]}"
+
+            val lastIsBlank = it.last().isBlank()
+
+            it[it.lastIndex] = it[it.lastIndex].trim() + " */"
+
+            it.mapIndexed { index, s ->
+                buildString {
+                    if (index != 0) append("\n")
+                    append(" ".repeat(indent))
+
+                    if (!(index == 0 || index == it.lastIndex && lastIsBlank)) {
+                        append(" *")
+                        if (s.isNotEmpty()) append(" ")
+                    }
+                    append(s)
+                }
+            }.joinToString("")
+        }
+
+/**
+ * Can retrieve the arguments of an inline- or block-tag.
+ * Arguments are split by spaces, unless they are in a block of "{}", "[]", "()", "<>", "`", """, or "'".
+ * Blocks "marks" are ignored if "\" escaped.
+ */
+fun String.getTagArguments(tag: String, numberOfArguments: Int): List<String> {
+    require("@$tag" in this) { "Could not find @$tag in $this" }
+    require(numberOfArguments > 0) { "numberOfArguments must be greater than 0" }
+
+    var content = this
+
+    // remove inline tag stuff
+    if (content.startsWith("{") && content.endsWith("}")) {
+        content = content.removePrefix("{").removeSuffix("}")
+    }
+
+    // remove leading spaces
+    content = content.trimStart()
+
+    // remove tag
+    content = content.removePrefix("@$tag").trimStart()
+
+    val arguments = buildList {
+        var currentBlock = ""
+        val blocksIndicators = mutableListOf<String>()
+
+        var escapeNext = false
+        for (char in content) {
+            when {
+                escapeNext -> escapeNext = false
+
+                char == '\\' -> escapeNext = true
+
+                size >= numberOfArguments - 1 -> Unit
+
+                char.isWhitespace() && blocksIndicators.isEmpty() -> {
+                    if (currentBlock.isNotBlank()) add(currentBlock)
+                    currentBlock = ""
+                }
+
+                char == '{' -> blocksIndicators += "{}"
```

Comment:
Usually, these string literals should be extracted to the constants at the top of the file, but for the God of readability better to keep it like that

Reply:
I think any performance gains are negligible. Plus, readability will definitely take a hit if I make it constants.
Sure things like ```const val `{` = '{'``` works, but ```const val `[` = '['``` is an illegal name. I think I'll keep the chars as is and make the Strings constants.

Reply:
so the stuff going in and out of `blocksIndicators` is constant :)